### DISCUSSION TITLE
Zod resolver and required fields

### DISCUSSION THREAD

**Discussion by wilson-webdev** [react-hook-form/react-hook-form#10041] [Q&A]
As part of a React app, it's pretty common to have a `FormField` wrapper for displaying labels, error messages etc. I have paired Zod with React Hook Form via the resolver. Is it possible to dynamically fetch whether a field is required based on the Zod schema?

For example, I have [this code sandbox](https://codesandbox.io/s/react-typescript-forked-s6ti26?file=/src/App.tsx) with a `FormFeld` component, it takes a prop called `isRequired` which renders a red asterisk if `true`. Of course, I could pass `isRequired` "manually", but I would like the schema to drive everything e.g. a small util like `isFieldRequired(key)` would be great, but I can't find anything in the docs that would help with this.

**Comment by brunocentanaro** [comment]
Hello! Did you find a solution?

**Reply by wilson-webdev** [reply]
I did find one option which is using the `isOptional()` helper, so you can write something like `schema.shape.firstName.isOptional()`. This only returns `true` if the prop has `.optional()` e.g. `firstName: z.string().optional()`.

**Reply by RakibulBh** [reply]
Need more help on your reply, when rendering the FormItem, do i do it to the field attribute ?

**Reply by Armadillidiid** [reply]
This fix doesn't scale when your component is used across multiple components, as it would require adding the schema to a context provider. It would be helpful if react-hook-form exposed an `isRequired` property on `ControllerRenderProps`.

**Reply by yashgo09** [reply]
Any solutions so far? I am also facing this issue.
